backend/src/main/java/com/javahelp/backend/accountdeletion/DeleteManager.java

```diff
@@ -0,0 +1,61 @@
+package com.javahelp.backend.accountdeletion;
+
+import com.javahelp.backend.data.IUserStore;
+import com.javahelp.model.user.User;
+import com.javahelp.model.user.UserPassword;
+
+import java.util.Arrays;
+
+/**
+ * A DeleteManager for an account deletion action.
+ */
+class DeleteManager {
+    /**
+     * The UserStore to use.
+     */
+    private final IUserStore iUserStore;
+
+    /**
+     * Constructs a DeleteManager instance.
+     *
+     * @param iUserStore: the UserStore used to store users.
+     */
+    protected DeleteManager(IUserStore iUserStore) {
+        this.iUserStore = iUserStore;
+    }
+
+    /**
+     * A helper method used to verify whether the entered password matches the password
+     * stored in the database.
+     *
+     * @param userID: the userID of the user.
+     * @param userPassword: the password entered by the user.
+     * @return whether the password entered matches the password in the database.
+     */
+    private boolean verify(String userID, UserPassword userPassword) {
+        UserPassword dbPassword = iUserStore.readPassword(userID);
+        return Arrays.equals(dbPassword.getHash(), userPassword.getHash());
+    }
+
+    /**
+     * Tries to delete the user with the given userID and the entered password.
+     * If the entered password does not match the password for this user in the database,
+     * the deletion is unsuccessful.
+     *
+     * Else, the user with the given userID will be deleted from the database.
+     *
+     * @param userID: the userID of the user.
+     * @param userPassword: the password entered by the user.
+     * @return a DeleteResult instance encoding whether the deletion is successful.
+     */
+    protected DeleteResult delete(String userID, UserPassword userPassword) {
+        User user = iUserStore.read(userID);
+
+        if (verify(userID, userPassword)) {
+            iUserStore.delete(userID);
+            return new DeleteResult(true, user);
+        }
+
+        return new DeleteResult(false, user);
```

Comment:
If the user is deleted, why do we keep the user in DeleteResult? Shouldn't DeleteResult return only success or does keeping the user in DeleteResult useful for us?

Reply:
I think it will be useful for later if we want to display any message in the frontend such as "User with {userID} has been deleted" or anything like that.